imap-codec/examples/common/common.rs

```diff
@@ -45,5 +45,12 @@ fn read_line(prompt: &str, role: Role) -> String {
 
     print!("{RESET}");
 
-    line.replace('\n', "\r\n")
+    // Ensure CRLF line endings
+    // (could be LF on Unix or CRLF on Windows)
+    line.lines()
```

Comment:
I... think I see what happens here. But maybe something like... 

```rust
if !line.ends_with("/r/n) {
    line.replace()
}
```

... would be easier to understand? (In case the feature quirk idea turns out to introduce too much hassle.)

Reply:
Okay, let's dig deeper into this to find the optimal solution ...

The current implementation `line.replace('\n', "\r\n")` will change _all `LF` line endings_ in `line` to `CRLF` (also, unfortunately, incorrectly change `CRLF` to `CRCRLF`). My proposed change would change _all line endings_ (`LF` or `CRLF`) in `line` to `CRLF`.

However, looking into the documentation of [BufRead::read_line](https://doc.rust-lang.org/stable/std/io/trait.BufRead.html#method.read_line) there should only be one `LF` at the end of the string (which might belong to a `CRLF`) - or none on EOF if stdin has been closed. Therefore, we could indeed simplify this to only check the end of the string.
(As is turns out, EOF is currently also not correctly handled and should probably be interpreted as `exit` instead of infinitely looping on empty input. 😉 I will add a second commit fixing this as well, after we have decided on an implementation here.)

Basically, in this location we want to do something like this to handle the three possible cases:
```rust
    if line.ends_with("\r\n") {
        return line;
    }
    if line.ends_with("\n") {
        line.pop();
    }
    line + "\r\n"
```

Do we care about trailing spaces to be preserved? Otherwise, we could also do something like:
```rust
line.trim_end().to_string() + "\r\n"
```
However, in contrast to the previous snippets, this would copy the string once after trimming.

What are your thoughts on how to proceed here?

Reply:
Great analysis, thanks! I like the first solution you propose!

> Do we care about trailing spaces to be preserved?

Yes. You could for example have a greeting ...

```
* OK Hello___\r\n
```

... that should become ...

```rust
Greeting {
  kind: Ok,
  code: None,
  text: "Hello___",
}
```

... with "_" meaning space here.

Reply:
Thanks for your input. I have now implemented the first solution and also added a fix for handling EOF correctly.